File: week3/data_reduction.r
# PCA Example
# Perform PCA
pca_result <- prcomp(scaled_data, center = TRUE, scale. = TRUE)

# Extract principal components
reduced_data <- pca_result$x[, 1:2]

# Feature selection using correlation
library(caret)

# Calculate correlation matrix
correlation_matrix <- cor(numerical_data)

# Find highly correlated features
high_correlation <- findCorrelation(correlation_matrix, cutoff = 0.8)
features_to_remove <- names(numerical_data)[high_correlation]

File: week3/data_reduction.py
# PCA Example
from sklearn.decomposition import PCA

# Create PCA object with desired components
pca = PCA(n_components=2)

# Fit and transform data
reduced_data = pca.fit_transform(scaled_data)

# Feature selection using correlation
import pandas as pd
import numpy as np

# Calculate correlation matrix
corr_matrix = df.corr().abs()

# Select upper triangle of correlation matrix
upper = corr_matrix.where(np.triu(np.ones(corr_matrix.shape), k=1).astype(bool))

# Find features with correlation greater than 0.8
to_drop = [column for column in upper.columns if any(upper[column] > 0.8)]
